apps/desktop/src/lib/url/gitUrl.ts

```diff
@@ -4,20 +4,22 @@ export type RepoInfo = {
 	source: string;
 	name: string;
 	owner: string;
+	resource: string;
```

Comment:
Wouldn't it be misleading to use an optional attribute for the `resource`? 

The `gitUrlParse` function seems to always return a `resource` attribute when the url is valid, even if it's an empty 'resource' it's not undefined.

Reply:
Sorry, I forgot what it represents, and sorry for the back and forth here. I think.. we should only have one field for url, instead of calling it source and resource. If you agree, would you mind creating a type and replacing the one we borrow from the parsing lib?

Reply:
Back and forth is how good software is made 😄 

You mean removing the `resource` and `source` from the `RepoInfo` type and have only a `url` attribute which will be the value of `resource` (the full url)?

I think the `source` is only used in `DefaultGitHostFactory` class, so should be easy to replace it.

Reply:
Yes, exactly! I would probably go with `domain` over `url` since it is likely to have additional things like scheme, path, etc. Also, my previous message was probably confusing because I forgot we have our own `RepoInfo` type, I thought we were using the one returned from `git-url-parse`.

Reply:
@mtsgrd Done!

Removed all references of `source` and `resource` in favour or using domain.